File: leetcode/python/0242_valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False
        freq = {}
        for i in range(len(s)):
            freq[s[i]] = freq.get(s[i], 0) + 1
            freq[t[i]] = freq.get(t[i], 0) - 1
        
        for k, v in freq.items():
            if v != 0: return False
        return True

File: leetcode/cpp/0242_valid_anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }
        
        unordered_multiset<char> lettersS;
        
        for (int i = 0; i < s.size(); i++) {
            lettersS.insert(s[i]);
        }
        
        for (int i = 0; i < t.size(); i++) {
            if (lettersS.find(t[i]) != lettersS.end()) {
                lettersS.erase(lettersS.find(t[i]));
            }
            else {
                return false;
            }
        }
        
        return true;
    }
};